galpy/potential/SCFPotential.py

```diff
@@ -685,6 +714,108 @@ def integrand(xi, costheta):
         Acos[:,:,0] = 2*I**-1 * integrated*constants
         
         return Acos, Asin
+    
+def scf_compute_coeffs_nbody(pos,mass,N,L,a=1., radial_order=None, costheta_order=None, phi_order=None):
+        
+        """        
+        NAME:
+
+           scf_compute_coeffs
+
+        PURPOSE:
+
+           Numerically compute the expansion coefficients for a given triaxial density
+
+        INPUT:
+
+           pos - Positions of particles
+           
+           m - mass of particles
+
+           N - size of the Nth dimension of the expansion coefficients
+
+           L - size of the Lth and Mth dimension of the expansion coefficients
+           
+           a - parameter used to shift the basis functions
+
+           radial_order - Number of sample points of the radial integral. If None, radial_order=max(20, N + 3/2L + 1)
+
+           costheta_order - Number of sample points of the costheta integral. If None, If costheta_order=max(20, L + 1)
+
+           phi_order - Number of sample points of the phi integral. If None, If costheta_order=max(20, L + 1)
+
+        OUTPUT:
+
+           (Acos,Asin) - Expansion coefficients for density dens that can be given to SCFPotential.__init__
+
+        HISTORY:
+
+           2020-11-18 - Written - Morgan Bennett
+
+        """ 
+        
+        n = numpy.arange(0,N)
+        l = numpy.arange(0,L)
+        m = numpy.arange(0,L)
+
+        r = numpy.sqrt(pos[0]**2+pos[1]**2+pos[2]**2)
+        phi = numpy.arctan2(pos[1],pos[0])
+        costheta = pos[2]/r
+        '''
+        Plm= numpy.zeros([len(costheta),1,L,L])
+        for i,ll in enumerate(l):
+            for j,mm in enumerate(m):
+                Plm[:,0,j,i]= lpmv(ll,mm,costheta)
+
+        cosmphi= numpy.cos(phi[:,None]*m[None,:])[:,None,None,:]
+        sinmphi= numpy.sin(phi[:,None]*m[None,:])[:,None,None,:]
+        
+        Ylm= (numpy.sqrt((2.*l[:,None]+1)*gamma(l[:,None]-m[None,:]+1)/gamma(l[:,None]+m[None,:]+1))[None,None,:,:]*Plm)[None,:,:,:,:]*numpy.array([cosmphi,sinmphi])
+        Ylm= numpy.nan_to_num(Ylm)
+
+        C= [[gegenbauer(nn,2.*ll+1.5) for ll in l] for nn in n] 
+        Cn= numpy.zeros((1,len(r),N,L,1))
+        for i in range(N):
+            for j in range(L):
+                Cn[0,:,i,j,0]= C[i][j]((r/a-1)/(r/a+1))
```

Comment:
Likewise, ``eval_ gegenbauer`` is not an astropy-supported ufunc, so the following is necessary.

``Cn[0,:,i,j,0]= C[i][j]( u.Quantity((r/a-1)/(r/a+1), copy=False).to_value(u.one) )``